src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnectorConfig.java

```diff
@@ -125,6 +125,11 @@ public class SnowflakeSinkConnectorConfig {
   public static final boolean SNOWPIPE_FILE_CLEANER_FIX_ENABLED_DEFAULT = false;
   public static final int SNOWPIPE_FILE_CLEANER_THREADS_DEFAULT = 1;
 
+  // Whether to close streaming channels in parallel.
+  public static final String SNOWPIPE_STREAMING_CLOSE_CHANNELS_IN_PARALLEL =
+      "snowflake.streaming.closeChannelsInParallel.enabled";
+  public static final boolean SNOWPIPE_STREAMING_CLOSE_CHANNELS_IN_PARALLEL_DEFAULT = false;
```

Comment:
This doesn't feel like a configuration should be exposed to customer. As a customer, I don't know whether I should close the channels in parallel or not. In fact, I don't really care, I just want my system to run without any failure. WDYT?

Reply:
agreed but I think we want to try this out for customers who has a lot of partitions in their single task. so enabling by default will not benefit a lot of customers 
- also its good to have param protection. 
- We should stress test this and slow roll out in couple of releases.

Reply:
What's the risk of having it default true and then getting rid of it in a future release?

Reply:
@sfc-gh-tzhang 
This is not a publicly announced parameter. It's going to behave as a knob that's going to allow tho customer to easily roll the change back in case something goes wrong. Exactly what @sfc-gh-japatel wrote.

@sfc-gh-xhuang 
Imo it's better to be safe than sorry.  I'm not sure how the parallelization is going to work on the customer environments as channel closing is assigned to a shared `ForkJoinPool` that seems to exist as a single instance and shared across all tasks on the same host.

I suggest we approach the "change rollout" parameters the following way:
1. Protect the change with a parameter and disable it by default. If some customers experience problems that can be solved by the change, they can turn it on. 
1. Enable the change by default. If some customers experience problems that cause by the change, they can turn it off and communicate to us.
1. Get rid of the older implementation.

Reply:
Understood but I was just thinking that there is not much difference/value between steps 1 and 2 but yes we can be a bit safer this way.

Reply:
Would this still be a behavior change after we finish step 3? This configuration is gone and customer's KC won't be able to start, then they will need to get the logs to understand why. Assuming they know how to look into the log, then they realize that the configuration is being removed and they will probably need to reach out to Snowflake to understand why it's being removed?

It seems to me a better way is to do a beta release with the fix so customer can try it (or just provide a private jar)? Other customers won't use this version since it's a beta version.

Reply:
@sfc-gh-tzhang I don't see any schema validation for configuration properties. We will just stop respecting the `snowflake.streaming.closeChannelsInParallel.enabled` config key, but no errors should be observed. So, I suppose there will be no behavioral change afterwards.